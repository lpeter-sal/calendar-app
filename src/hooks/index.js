export * from './useUiStore';
export * from './useCalendarSotre';
export * from './useForms';
export * from './useAuthStore';