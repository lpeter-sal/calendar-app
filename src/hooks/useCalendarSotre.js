import { useDispatch, useSelector } from "react-redux"
import { onAddNewEvent, onDeleteEvent, onSetActiveEvent, onUpdateEvent } from "../store/calendar/calendarSlice";

export const useCalendarSotre = () => {

    const dispatch = useDispatch();
    const { events, activeEvent } = useSelector( state => state.calendar );

    const setActiveEvent = ( calendarEvent ) => {
        dispatch( onSetActiveEvent( calendarEvent ) );
    }
  
    const startSavingEvent = async( calendarEvent ) => {
        


        if( calendarEvent._id ) {
            //Update

            dispatch( onUpdateEvent({...calendarEvent}) );
        } else {
            //Create
            dispatch( onAddNewEvent({ ...calendarEvent, _id: new Date().getTime() }) );
        }
    }
    
    const startDeletingEvent = () => {



        dispatch( onDeleteEvent() );
    }
  
  
    return {
            // Properties
            activeEvent,
            events,
            hasEventSelected: !!activeEvent,

            // Methods
            setActiveEvent,
            startDeletingEvent,
            startSavingEvent,
    }
    
}
