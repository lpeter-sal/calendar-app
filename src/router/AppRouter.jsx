import { Navigate, Route, Routes } from "react-router-dom"
import { AuthRoutes } from "../auth/router/AuthRoutes";
import { CalendarRouter } from "../calendar/router/CalendarRouter";


export const AppRouter = () => {

  const authStatus = 'not-authenticated'; //'not-authenticated'



  return (
    <Routes>
      
        {
          ( authStatus === 'authenticated')
          ? <Route path="/*" element={ <CalendarRouter /> } />
          : <Route path="/auth/*" element ={ <AuthRoutes /> } />
        }

        <Route path="/*" element={ <Navigate to="/auth/login" /> } />
        
    </Routes>
  )
}
