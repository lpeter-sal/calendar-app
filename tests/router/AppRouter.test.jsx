import { render, screen } from "@testing-library/react";
import { useAuthStore } from "../../src/hooks/useAuthStore";
import { AppRouter } from "../../src/router/AppRouter";
import { MemoryRouter } from "react-router-dom";

jest.mock('../../src/hooks/useAuthStore');

jest.mock('../../src/calendar/router/CalendarRouter', () => ({
    CalendarRouter: () => <h1>CalendarRoute</h1>
}));



describe('Pruebas en <AppRouter />', () => { 

    const mockCheckAuthToken = jest.fn();

    beforeEach( () => jest.clearAllMocks() );


    test('Debe de mostrar la pantalla de carga y llamar AuthCheckToken', () => { 

        useAuthStore.mockReturnValue({
            status: 'checking',
            checkAuthToken: mockCheckAuthToken
        });
        
        render( <AppRouter /> )
        
        expect(screen.getByText('Cargando...')).toBeTruthy();
        expect(mockCheckAuthToken).toHaveBeenCalled();
    });

    test('Debe de mostrar el login en caso de no estar autenticado', () => { 

        useAuthStore.mockReturnValue({
            status: 'not-authenticated',
            checkAuthToken: mockCheckAuthToken
        });
        
        const { container } = render( 
            <MemoryRouter initialEntries={ ['/auth/algo/otracosa'] }>
                <AppRouter />
            </MemoryRouter> 
        );

        expect(screen.getByText('LOGIN')).toBeTruthy();
        expect( container ).toMatchSnapshot();
    });

    test('Debe de mostrar el calendario si estamos autenticados', () => { 

        useAuthStore.mockReturnValue({
            status: 'authenticated',
            checkAuthToken: mockCheckAuthToken
        });
        
       render( 
            <MemoryRouter>
                <AppRouter />
            </MemoryRouter> 
        );

        expect( screen.getByText('CalendarRoute')).toBeTruthy();
    });
    


});