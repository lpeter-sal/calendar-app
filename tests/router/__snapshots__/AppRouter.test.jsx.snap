// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> Debe de mostrar el login en caso de no estar autenticado 1`] = `
<div>
  <div
    class="cont-body"
  >
    <div
      class="wrapper"
    >
      <form>
        <h1>
          CALENDAR APP
        </h1>
        <h3>
          LOGIN
        </h3>
        <div
          class="input-box"
        >
          <input
            name="loginEmail"
            placeholder="Email"
            type="text"
            value=""
          />
          <i
            class="bx bxs-envelope"
          />
        </div>
        <div
          class="input-box"
        >
          <input
            name="loginPassword"
            placeholder="Password"
            type="password"
            value=""
          />
          <i
            class="bx bxs-lock-alt"
          />
        </div>
        <input
          class="btn"
          type="submit"
          value="Login"
        />
        <div
          class="register-link"
        >
          <p>
            Don't have an account?
            <a
              href="/auth/register"
            >
              Â  Register HERE!
            </a>
          </p>
        </div>
      </form>
    </div>
  </div>
</div>
`;
